File: core/src/com/qtech/bubbleblaster/effect/DefenseBoostEffect.java
package com.qtech.bubbleblaster.effect;

import com.qtech.bubbleblaster.common.AttributeMap;
import com.qtech.bubbleblaster.common.effect.Effect;
import com.qtech.bubbleblaster.common.effect.EffectInstance;
import com.qtech.bubbleblaster.common.entity.Attribute;
import com.qtech.bubbleblaster.common.entity.Entity;

public class DefenseBoostEffect extends Effect {
    public DefenseBoostEffect() {
        super();
    }

    @Override
    protected boolean canExecute(Entity entity, EffectInstance effectInstance) {
        return false;
    }

    @Override
    public void execute(Entity entity, EffectInstance effectInstance) {

    }

    @Override
    public AttributeMap getAttributeModifiers() {
        AttributeMap map = new AttributeMap();
        map.setBase(Attribute.DEFENSE, 1f);
        return map;
    }
}

File: src/main/kotlin/qtech/bubbles/effect/DefenseBoostEffect.kt
package qtech.bubbles.effect

import qtech.bubbles.common.AttributeMap
import qtech.bubbles.common.effect.Effect
import qtech.bubbles.common.effect.EffectInstance
import qtech.bubbles.common.entity.Attribute
import qtech.bubbles.entity.Entity

class DefenseBoostEffect : Effect() {
    override fun canExecute(entity: Entity, effectInstance: EffectInstance): Boolean {
        return false
    }

    override fun execute(entity: Entity, effectInstance: EffectInstance) {}
    override val attributeModifiers: AttributeMap
        get() {
            val map = AttributeMap()
            map.set(Attribute.DEFENSE, 1f)
            return map
        }
}